### PR TITLE
Add editor-position property

### LINKED ISSUE #167: Add option to show editor view on below the grid
- Editor, when opened, would cause grid to shrink on its height
- In small screens, editor opens in full screen

### PR DESCRIPTION
**Pull request by DiegoCardoso**
- Accepts one value: `bottom`
  - _empty string (default) works as overlay_
- Add demo showing the new feature
- Theme updated

Fixes #167

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-crud/169)

### CODE REVIEW

File: theme/lumo/vaadin-crud-styles.html [hunk 1]
```diff
@@ -104,6 +104,36 @@
...
+      }
+
+      @media (max-width: 600px), (max-height: 600px) {
+        :host([editor-position='bottom']) [part="editor"]:not([hidden]) {
+          position: fixed;
+          top: 0;
+          left: 0;
+          right: 0;
+          bottom: 0;
+          padding: 0;
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-crud-styles.html]: Not sure if it is needed as it's not affecting anything.

File: theme/lumo/vaadin-crud-styles.html [hunk 2]
```diff
@@ -104,6 +104,36 @@
...
+        }
+
+        :host([theme~="layout"]) [part="overlay"] {
+          height: 100vh;
+          width: 100vw;
+          border-radius: 0 !important;
+        }
+
+        :host([theme~="layout"]) [part="content"] {
+          flex: 1;
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-crud-styles.html]: I guess there is no content affected by this.

File: test/crud-test.html [hunk 3]
```diff
@@ -626,6 +626,39 @@
           crud.__save();
         });
       });
+
+      describe('editor-position', () => {
+        let crud;
+
+        beforeEach(done => {
+          crud = fixture('default');
+          Polymer.RenderStatus.afterNextRender(crud, done);
```
Comment (yuriy-fix) [inline test/crud-test.html]: Why the test should be async? Seems to be passing without this one.

File: src/vaadin-dialog-layout.html [hunk 4]
```diff
@@ -132,14 +154,34 @@
...
+                  this.__ariaLabel = Array.from(this.$.dialog.$.overlay.$.content.querySelectorAll('[slot=header]'))
+                    .reduce((prev, ele) => prev + ' ' + ele.textContent, '').trim());
+              }
             }
+
+            // Make sure to reset scroll position
+            this.$.scroller.scrollTop = 0;
+          }
+
+          _computeEditorOpened(opened, editor) {
```
Comment (yuriy-fix) [inline src/vaadin-dialog-layout.html]: Nit: the second param can be changed to smth like `editorPosition` :)
Not a blocker.

File: src/vaadin-crud.html [hunk 5]
```diff
@@ -376,13 +413,29 @@ <h3 slot="header">[[__editorHeader]]</h3>
...
 
+        __onEditorPositionChange(editorPosition, currentEditorPosition) {
+          if (CrudElement._isValidEditorPosition(editorPosition)) {
+            return;
+          }
+          this.editorPosition = '';
+        }
+
+        static _isValidEditorPosition(editorPosition) {
+          return ['bottom'].indexOf(editorPosition) != -1;
```
Comment (yuriy-fix) [inline src/vaadin-crud.html]: Should it work with the values like `b`, `bot`? Should we compare the strings over here?

**Comment by web-padawan** [comment]
Note regarding IE11 appearance: editor height does not seem to be 40%

## IE11

ie

## Edge

edge

**Pull request merged**